scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -474,15 +474,17 @@ def device_is_available(self, instance):
 
         # Select an available DUT with less failures
         for d in sorted(duts_found, key=lambda _dut: _dut.failures):
-            d.lock.acquire()
+            duts_shared_hw = [_d for _d in self.duts if _d.id == d.id]  # get all DUTs with the same id
+            [_d.lock.acquire() for _d in duts_shared_hw]
```

Comment:
```suggestion
            for _d in duts_shared_hw:
                _d.lock.acquire()
```

Reply:
thanks! I found it was also issued in check_compliance script. I updated it and other issues with whitespaces.